cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -110,3 +153,254 @@ func (pe *ProbeEvent) initOrErrorEvent() {
+					// get call failed
+				}
+
+				if parentBDAPI.Status.ClaimState != apis.BlockDeviceUnclaimed {
+					// device is in use, and the consumer is doing something
+					// do nothing
+					klog.V(4).Infof("parent device: %s is in use, device: %s can be ignored", parentBD.DevPath, bd.DevPath)
+					return nil
+				} else {
+					// the consumer created some partitions on the disk.
+					// So the parent BD need to be deactivated and partition BD need to be created.
+					// 1. deactivate parent
+					// 2. create resource for partition
+
+					pe.Controller.DeactivateBlockDevice(*parentBDAPI)
+					deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+					existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, deviceInfo.UUID)
+					err := pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
+					if err != nil {
+						klog.Error(err)
+						return err
+					}
+					return nil
+				}
+
+			}
+
+			if bd.DeviceAttributes.DeviceType != libudevwrapper.UDEV_PARTITION &&
+				len(bd.DependentDevices.Partitions) > 0 {
+				klog.V(4).Infof("device: %s has partitions: %+v", bd.DevPath, bd.DependentDevices.Partitions)
+				return nil
+			}
+
+			if len(bd.DependentDevices.Holders) > 0 {
+				klog.V(4).Infof("device: %s has holders: %+v", bd.DevPath, bd.DependentDevices.Holders)
+				return nil
+			}
+
+			klog.V(4).Infof("creating block device resource for device: %s with uuid: %s", bd.DevPath, bd.UUID)
+			deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+			existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, deviceInfo.UUID)
+			err := pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
+			if err != nil {
+				klog.Errorf("could not create block device resource, %+v", err)
+				return err
+			}
+			return nil
+		}
+
+		if err != nil {
+			klog.Errorf("querying etcd failed: %+v", err)
+			return err
+		}
+
+		if bdAPI.Status.ClaimState != apis.BlockDeviceUnclaimed {
+			klog.V(4).Infof("device: %s is in use. update the details of the blockdevice", bd.DevPath)
+			deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+			err = pe.Controller.PushBlockDeviceResource(bdAPI, deviceInfo)
+			if err != nil {
+				klog.Errorf("updating block device resource failed: %+v", err)
+				return err
+			}
+			return nil
+		}
+
+		klog.V(4).Infof("creating resource for device: %s with uuid: %s", bd.DevPath, bd.UUID)
+		deviceInfo := pe.Controller.NewDeviceInfoFromBlockDevice(&bd)
+		existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, deviceInfo.UUID)
+		err = pe.Controller.PushBlockDeviceResource(existingBlockDeviceResource, deviceInfo)
+		if err != nil {
+			klog.Errorf("creation of resource failed: %+v", err)
+			return err
+		}
+		return nil
+	} else {
+		klog.V(4).Infof("device: %s cannot be uniquely identified", bd.DevPath)
+		if len(bd.DependentDevices.Partitions) > 0 ||
```

Comment:
q_: if partitions are present, uuid will be definitely generated right?

Reply:
Yes, UUID will be generated. But whether resource will be created depends on whether the device has  any holders. Because if there are holders it means, we should not create for this device, but should create only for the holder device.